Mathlib/RingTheory/KrullDimension/Regular.lean

```diff
@@ -0,0 +1,164 @@
+  obtain ⟨r, hrm, hr⟩ := exists_minimalPrimes_le (mem_support_iff_of_finite.mp q.head.2)
+  let r : support R M := ⟨⟨r, minimalPrimes_isPrime hrm⟩, mem_support_iff_of_finite.mpr hrm.1.2⟩
+  have hr : r < q.head := lt_of_le_of_ne hr (fun h ↦ hn q.head.1.1 (by rwa [← h]) hp.2)
+  exact le_of_eq_of_le (by simp [q]) (le_iSup _ (q.cons r hr))
+
+theorem supportDim_quotSMulTop_succ_eq_of_notMem_minimalPrimes_of_mem_jacobson {x : R}
+    (hn : ∀ p ∈ (annihilator R M).minimalPrimes, x ∉ p) (hx : x ∈ (annihilator R M).jacobson) :
+    supportDim R (QuotSMulTop x M) + 1 = supportDim R M :=
+  le_antisymm (supportDim_quotSMulTop_succ_le_of_notMem_minimalPrimes hn)
+    (supportDim_le_supportDim_quotSMulTop_succ_of_mem_jacobson hx)
+
+theorem supportDim_quotSMulTop_succ_eq_supportDim_mem_jacobson {x : R} (reg : IsSMulRegular M x)
+    (hx : x ∈ (annihilator R M).jacobson) : supportDim R (QuotSMulTop x M) + 1 = supportDim R M :=
+  supportDim_quotSMulTop_succ_eq_of_notMem_minimalPrimes_of_mem_jacobson
+    (fun _ ↦ reg.notMem_of_mem_minimalPrimes) hx
+
+lemma _root_.ringKrullDim_quotient_span_singleton_succ_eq_ringKrullDim_of_mem_jacobson {x : R}
+    (reg : IsSMulRegular R x) (hx : x ∈ Ring.jacobson R) :
+    ringKrullDim (R ⧸ span {x}) + 1 = ringKrullDim R := by
+  have h := Submodule.ideal_span_singleton_smul x (⊤ : Ideal R)
+  simp only [smul_eq_mul, mul_top] at h
+  rw [ringKrullDim_eq_of_ringEquiv (quotientEquivAlgOfEq R h).toRingEquiv,
+    ← supportDim_quotient_eq_ringKrullDim, ← supportDim_self_eq_ringKrullDim]
+  exact supportDim_quotSMulTop_succ_eq_supportDim_mem_jacobson reg
+    ((annihilator R R).ringJacobson_le_jacobson hx)
+
+variable [IsLocalRing R]
+
+/-- If $M$ is a finite module over a Noetherian local ring $R$, then $\dim M \le \dim M/xM + 1$
+  for every $x$ in the maximal ideal of the local ring $R$. -/
+@[stacks 0B52 "the second inequality"]
+theorem supportDim_le_supportDim_quotSMulTop_succ {x : R} (hx : x ∈ maximalIdeal R) :
+    supportDim R M ≤ supportDim R (QuotSMulTop x M) + 1 :=
+  supportDim_le_supportDim_quotSMulTop_succ_of_mem_jacobson ((maximalIdeal_le_jacobson _) hx)
+
+@[stacks 0B52 "the equality case"]
+theorem supportDim_quotSMulTop_succ_eq_of_notMem_minimalPrimes_of_mem_maximalIdeal {x : R}
+    (hn : ∀ p ∈ (annihilator R M).minimalPrimes, x ∉ p) (hx : x ∈ maximalIdeal R) :
+    supportDim R (QuotSMulTop x M) + 1 = supportDim R M :=
+  supportDim_quotSMulTop_succ_eq_of_notMem_minimalPrimes_of_mem_jacobson hn <|
+    (maximalIdeal_le_jacobson _) hx
+
+theorem supportDim_quotSMulTop_succ_eq_supportDim {x : R} (reg : IsSMulRegular M x)
+    (hx : x ∈ maximalIdeal R) : supportDim R (QuotSMulTop x M) + 1 = supportDim R M :=
+  supportDim_quotSMulTop_succ_eq_supportDim_mem_jacobson reg ((maximalIdeal_le_jacobson _) hx)
+
+@[stacks 00KW]
+lemma _root_.ringKrullDim_quotient_span_singleton_succ_eq_ringKrullDim {x : R}
+    (reg : IsSMulRegular R x) (hx : x ∈ maximalIdeal R) :
```

Comment:
```suggestion
    (reg : x ∈ R⁰) (hx : x ∈ maximalIdeal R) :
```
I think we use this for rings. And this gives you `dim R[[X]] = dim R + 1` right? Can you add this or will this be in a follow up PR? Also how far are we from `dim R[X] = dim R + 1`?

Reply:
Sure